config/routes.rb

```diff
@@ -20,34 +20,33 @@ def matches?(request)
     resources :live_stream, only: %i[index update] do
       post "publish", to: "live_stream#publish"
     end
-  end
 
-  resources :coronavirus_pages,
-            path: "coronavirus",
-            only: %i[index show update],
-            param: :slug,
-            module: :coronavirus do
-    get "prepare", on: :member
-    get "discard", on: :member
-    post "publish", to: "coronavirus_pages#publish"
-    resources :sub_sections do
-      collection do
-        get "reorder", to: "reorder_sub_sections#index"
-        put "reorder", to: "reorder_sub_sections#update"
+    resources :pages,
+              path: "",
+              only: %i[index show update],
+              param: :slug do
+      get "prepare", on: :member
+      get "discard", on: :member
```

Comment:
Yup, `on: :member` treats the route as one that is intended for a single page whereas the default is to treat the route as a fresh resource within a page - it also treats the route as a plural i.e. `coronavirus_pages_publish` compared to the singular `publish_coronavirus_page`

To be honest, I only went for `on: :member` as the path of least resistance since we already had prepare and discard set-up that way. and didn't give it much wider thought Your point makes sense about consistent param however unfortunately doing that wouldn't actually make the params consistent as show and update would still have a `slug` parameter. So with a little thought I think `on: :member` is the best option and is probably most consistent with Rails routing intentions.

I did try get the params to be all consistent before but failed to do so. I wondered if I set the param on line 27 as `page_slug` whether that would apply it to everything. However that ended up with the nested routes having `page_page_slug` 🤦